qtawesome/iconic_font.py

```diff
@@ -56,17 +67,35 @@ def paint(self, iconic, painter, rect, mode, state, options):
     def _paint_icon(self, iconic, painter, rect, mode, state, options):
         """Paint a single icon."""
         painter.save()
-        color, char = options['color'], options['char']
-
-        if mode == QIcon.Disabled:
-            color = options.get('color_disabled', color)
-            char = options.get('disabled', char)
-        elif mode == QIcon.Active:
-            color = options.get('color_active', color)
-            char = options.get('active', char)
-        elif mode == QIcon.Selected:
-            color = options.get('color_selected', color)
-            char = options.get('selected', char)
+        color = options['color']
+        char = options['char']
+
+        if state == QIcon.On:
+            if mode == QIcon.Normal:
+                color = options['color_on']
+                char = options['on']
+            elif mode == QIcon.Disabled:
+                color = options['color_on_disabled']
+                char = options['on_disabled']
+            elif mode == QIcon.Active:
+                color = options['color_on_active']
+                char = options['on_active']
+            elif mode == QIcon.Selected:
+                color = options['color_on_selected']
+                char = options['on_selected']
+        elif state == QIcon.Off:
+            if mode == QIcon.Normal:
+                color = options['color_off']
+                char = options['off']
+            elif mode == QIcon.Disabled:
+                color = options['color_off_disabled']
+                char = options['off_disabled']
+            elif mode == QIcon.Active:
+                color = options['color_off_active']
+                char = options['off_active']
+            elif mode == QIcon.Selected:
+                color = options['color_off_selected']
+                char = options['off_selected']
```

Comment:
How about using dicts here instead of those elif ? This part could be written as `color, char = color_options[state][mode]`, and you'd have to fill the dict somewhere else. It would simplify the logic of the code inside the function.